api/src/models/ImageFile.ts

```diff
@@ -56,7 +56,7 @@ class ImageFile {
         var path = require('path');
         var dir = path.dirname(this.filename);
         var filename = this.basename(this.filename);
-        return dir + "/" + filename + "." + size + "." + extension.toLowerCase();
+        return dir + "/" + filename + "/" + size + "/" + extension.toLowerCase();
```

Comment:
For consistency with the Ruby, I think you want:

`return dir + "/" + filename + "/" + size + "/" + filename + "." + extension.toLowerCase();`

(It uses the filename both as a directory for grouping things together, and as the actual filename for the generated derivative... I don't know that all this is strictly necessary, but if we want to keep it as it is, that other filename should be added back in).